### DISCUSSION TITLE
zh-hant/chapter_introduction/what_is_dsa/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#1330] [Announcements]
# zh-hant/chapter_introduction/what_is_dsa/

動畫圖解、一鍵執行的資料結構與演算法教程

https://www.hello-algo.com/zh-hant/chapter_introduction/what_is_dsa/

**Comment by keeper0811** [comment]
以下我的理解

#演算法（Algorithm）
演算法就像是拼裝積木時的說明書；說明書會一步一步告訴你，應該先拿哪個積木，再拿哪個積木，然後怎麼把它們組裝在一起，最後完成一個很棒的作品。

具體例子，說明書會告訴你想拼一輛車子步驟如下：
步驟一：先拿一塊大的底板。
步驟二：把四個輪子裝在底板的四個角上。
步驟三：接著拿一塊長方形的積木裝在輪子上，當作車身。
步驟四：最後裝上一個方形的積木當作駕駛室。
這個組裝過程就是一個演算法，它告訴你怎麼一步步完成你的作品。

#資料結構（Data Structure）
資料結構就像是你如何整理和存放這些積木的方式。當你在拼裝積木時，積木可能有很多種顏色、形狀和大小。如果你把它們亂七八糟地堆在一起，很難找到你需要的那塊。但如果你把積木按照顏色或形狀分類，放進不同的盒子裡，那麼當你需要一塊特定的積木時，就可以很快找到。

資料結構就是用來整理和管理資料的方法。例如，你可以把積木放在一個盒子裡（這就是一種簡單的資料結構），或者把積木按照顏色放進不同的小盒子裡（這就是更複雜的資料結構）。

具體例子：
1.陣列（Array）：就像是一排排整齊放好的積木。每個積木都有自己的位置編號，這樣你可以很快地找到某一個特定位置的積木。
2.鏈結串列（Linked List）：想像每塊積木上都有一個標籤，標籤上寫著下一塊積木的位置。你可以從第一塊開始，一直按照標籤找到下一塊，直到你找到你需要的那塊積木。

#總結一下：
演算法是告訴你怎麼一步一步做事情的計劃，而資料結構是你怎麼整理和存放資料的方法；就像拼裝積木一樣，演算法是拼一輛車的說明書，資料結構是存放還沒拼的積木的方式。
